### PR TITLE
kill and reap ProxyCommand

### PR DESCRIPTION
**Pull request by daaku**
Using the scopeguard crate ensures the child process (if any) spawned for the ProxyCommand is cleaned up under all the various error scenarios (such as auth failures etc), as well as in the normal successful use case. This ensures killing it if necessary, and then waiting for it.

This is a fix for #5479.

### CODE REVIEW

File: wezterm-ssh/Cargo.toml [hunk 1]
```diff
@@ -39,6 +39,7 @@ wezterm-uds = { path = "../wezterm-uds" }
 
 # Not used directly, but is used to centralize the openssl vendor feature selection
 async_ossl = { path = "../async_ossl" }
+scopeguard = "1.2.0"
```
Comment (wez) [inline wezterm-ssh/Cargo.toml:42]: while scopeguard is handy in a pinch, I'm not in favor of it here for a couple of reasons:

* It is an additional external dependency
* You still have to remember to initiate the scope guard, and that is required in multiple places in this PR

What I'd prefer to see as a solution for this is for `connect_to_host` to return a little wrapper around the Child:

```rust
struct KillOnDropChild(Child);

impl std::ops::Deref for KillOnDropChild { ... }
impl std::ops::DerefMut for KillOnDropChild { ... }
impl Drop for KillOnDropChild {
   fn drop(&mut self) {
       self.0.kill();
       self.0.wait();
   }
}
```

**Review by wez — CHANGES_REQUESTED** [review]
Thanks for this!
I have a slightly different proposal; would you mind updating to match it?

**Comment by daaku** [comment]
Sounds reasonable, I'll update with the requested changes.

**Comment by wez** [comment]
Thank you!

**Pull request merged**